docs/src/snap/howto/ceph.md

```diff
@@ -0,0 +1,339 @@
+
+Consider this important note from the Ceph documentation:
+
+> The provisioner and node plugin YAMLs will, by default, pull the development
+> release of the ceph-csi container (quay.io/cephcsi/cephcsi:canary). The YAMLs
+> should be updated to use a release version container for production workloads.
+
+## Create a StorageClass
+
+You are ready to create a ceph-csi StorageClass.
+
+```
+cat <<EOF > csi-rbd-sc.yaml
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: csi-rbd-sc
+provisioner: rbd.csi.ceph.com
+parameters:
+   clusterID: b9127830-b0cc-4e34-aa47-9d1a2e9949a8
+   pool: kubernetes
+   imageFeatures: layering
+   csi.storage.k8s.io/provisioner-secret-name: csi-rbd-secret
+   csi.storage.k8s.io/provisioner-secret-namespace: default
+   csi.storage.k8s.io/controller-expand-secret-name: csi-rbd-secret
+   csi.storage.k8s.io/controller-expand-secret-namespace: default
+   csi.storage.k8s.io/node-stage-secret-name: csi-rbd-secret
+   csi.storage.k8s.io/node-stage-secret-namespace: default
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+mountOptions:
+   - discard
+EOF
+```
+
+Then apply:
+
+```
+kubectl apply -f csi-rbd-sc.yaml
+```
+
+## Create a Persistant Volume Claim (PVC) for a RBD-backed file-system
+
+This PVC will allow users to request RBD-backed storage.
+
+```
+cat <<EOF > pvc.yaml
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: rbd-pvc
+spec:
+  accessModes:
+    - ReadWriteOnce
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: csi-rbd-sc
+EOF
+```
+
+Then apply:
+
+```
+kubectl apply -f pvc.yaml
+```
+
+## Create a pod that binds to the Rados Block Device PVC
+
+Finally, create a pod configuration that uses the RBD-backed PVC.
+
+```
+cat <<EOF > pod.yaml
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: csi-rbd-demo-pod
+spec:
+  containers:
+    - name: web-server
+      image: nginx
+      volumeMounts:
+        - name: mypvc
+          mountPath: /var/lib/www/html
+  volumes:
+    - name: mypvc
+      persistentVolumeClaim:
+        claimName: rbd-pvc
+        readOn
+EOF
+```
+
+Then apply:
+
+```
+kubectl apply -f pod.yaml
+```
+
+## Verify that the pod is using the RBD PV
+
+To verify that the csi-rbd-demo-pod is indeed using a RBD Persistant Volume, run
+the following commands, you should see information related to attached volumes
+in both of their outputs:
+
+```
+kubectl describe pvc rbd-pvc
+
+kubectl describe pod csi-rbd-demo-pod
+```
+
+Congratulations! By following this guide, you've set up a basic yet reliable
+persistent storage solution for your Kubernetes cluster. To further enhance and
+prepare your cluster for production use, we recommend reviewing the official
+Ceph documentation: [Intro to Ceph].
+
+<!-- LINKS -->
+
+[Ceph]: https://ceph.com/
+
+[getting-started-guide]:
+../tutorial/getting-started.md[block-devices-and-kubernetes]:
+https://docs.ceph.com/en/latest/rbd/rbd-kubernetes/ [placement groups]:
```

Comment:
these links seem to have been mangled somehow - that's why I word wrap on a paragraph by paragrah basis!

Reply:
Can't win at this !